SteamKit2/SteamKit2/Util/VZipUtil.cs

```diff
@@ -45,26 +48,41 @@ public static byte[] Decompress(byte[] buffer)
                 throw new Exception( "Expecting VZipFooter at end of stream" );
             }
 
+            if ( destination.Length < sizeDecompressed )
+            {
+                throw new ArgumentException( "The destination buffer is smaller than the decompressed data size.", nameof( destination ) );
+            }
+
             // jump back to the beginning of the compressed data
             ms.Position = compressedBytesOffset;
 
             SevenZip.Compression.LZMA.Decoder decoder = new SevenZip.Compression.LZMA.Decoder();
 
-            decoder.SetDecoderProperties( properties );
+            // If the value of dictionary size in properties is smaller than (1 << 12),
+            // the LZMA decoder must set the dictionary size variable to (1 << 12).
+            var windowBuffer = ArrayPool<byte>.Shared.Rent( Math.Max( 1 << 12, ( int )dictionarySize ) );
+
+            try
+            {
+                decoder.SteamKitSetDecoderProperties( propertyBits, dictionarySize, windowBuffer );
 
-            var outData = new byte[ sizeDecompressed ];
-            using MemoryStream outStream = new MemoryStream( outData );
-            decoder.Code( ms, outStream, sizeCompressed, sizeDecompressed, null );
+                using MemoryStream outStream = new MemoryStream( destination );
+                decoder.Code( ms, outStream, sizeCompressed, sizeDecompressed, null );
+            }
+            finally
+            {
+                ArrayPool<byte>.Shared.Return( windowBuffer );
+            }
 
-            if ( Crc32.HashToUInt32( outData ) != outputCRC )
+            if ( verifyChecksum && Crc32.HashToUInt32( destination.AsSpan()[ ..sizeDecompressed ] ) != outputCRC )
```

Comment:
The depot chunk itself has a checksum, so I don't see a reason to verify it here.